services/user.py

```diff
@@ -0,0 +1,46 @@
+from django.contrib.auth import get_user_model
+from db.models import User
+
+
+def create_user(username: str,
+                password: str,
+                email=None,
+                first_name=None,
+                last_name=None
+                ) -> User:
+    user = get_user_model().objects.create_user(username=username)
+    user.set_password(password)
+
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+
+    user.save()
+    return user
+
+def get_user(user_id: int) -> User:
+    return get_user_model().objects.get(pk=user_id)
+
+def update_user(user_id: int,
+                username: str = None,
+                password: str = None,
+                email: str = None,
+                first_name: str = None,
+                last_name: str = None) -> None:
+    user = get_user(user_id)
+
+    if username:
+        user.username = username
+    if password:
+        user.set_password(password)
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+
+    user.save()

```

Comment:
You are calling `save` method here which is not necessary. `set_password` method already saves the user object in the database. You only need to call `save` method if you have made changes to the user object after calling `set_password` method. This is a code quality issue.